Issue in TF record coders

There seems to be some kind of issue with TF record coders, there has been a couple of reports of issues with coder discovery, and I myself saw random failures like:

```
[info]   java.lang.IllegalArgumentException: requirement failed: Invalid masked CRC32 of length
[info]   at scala.Predef$.require(Predef.scala:277)
[info]   at com.spotify.scio.io.TFRecordCodec$.read(TFRecordCodec.scala:45)
```

or

```
Caused by: java.util.zip.ZipException: invalid stored block lengths
INFO:     at java.util.zip.InflaterInputStream.read(InflaterInputStream.java:164)
INFO:     at org.apache.commons.compress.compressors.deflate.DeflateCompressorInputStream.read(DeflateCompressorInputStream.java:69)
INFO:     at com.spotify.scio.io.TFRecordCodec$.readFully(TFRecord.scala:221)
INFO:     at com.spotify.scio.io.TFRecordCodec$.read(TFRecord.scala:197)
INFO:     at com.spotify.scio.io.TFRecordReader.readNextRecord(TFRecord.scala:103)
```

--- a/scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TFRecordCodec.scala
+++ b/scio-tensorflow/src/main/scala/com/spotify/scio/tensorflow/TFRecordCodec.scala
@@ -94,20 +94,21 @@ private object TFRecordCodec {
   private def hashBytes(x: Array[Byte]): Int = mask(crc32c.hashBytes(x).asInt())
 
   private def isGzipInputStream(pushback: PushbackInputStream): Boolean = {
-    val headerBytes = Array.ofDim[Byte](2)
-    pushback.read(headerBytes)
+    val b1 = pushback.read()
+    val b2 = pushback.read()
+    if (b2 != -1) pushback.unread(b2)
+    if (b1 != -1) pushback.unread(b1)
     val zero: Byte = 0x00
-    val header = Ints.fromBytes(zero, zero, headerBytes(1), headerBytes(0))
-    pushback.unread(headerBytes)
-    header == GZIPInputStream.GZIP_MAGIC
+    val header = Ints.fromBytes(zero, zero, b2.toByte, b1.toByte)
+    (b1 != -1 && b2 != -1) && header == GZIPInputStream.GZIP_MAGIC
   }
 
   private def isInflaterInputStream(pushback: PushbackInputStream): Boolean = {
     val b1 = pushback.read()
     val b2 = pushback.read()
-    pushback.unread(b2)
-    pushback.unread(b1)
-    b1 == 0x78 && (b1 * 256 + b2) % 31 == 0
+    if (b2 != -1) pushback.unread(b2)
+    if (b1 != -1) pushback.unread(b1)
+    (b1 != -1 && b2 != -1) && (b1 == 0x78 && (b1 * 256 + b2) % 31 == 0)
   }
 
 }

